pkg/util/kube/vmsnapshot.go

```diff
@@ -170,3 +169,19 @@ func PatchSnapshotSuccessStatus(
 
 	return nil
 }
+
+func ConvertPowerState(logger logr.Logger,
+	powerState vimtypes.VirtualMachinePowerState) vmopv1.VirtualMachinePowerState {
+	switch powerState {
+	case vimtypes.VirtualMachinePowerStatePoweredOn:
+		return vmopv1.VirtualMachinePowerStateOn
+	case vimtypes.VirtualMachinePowerStatePoweredOff:
+		return vmopv1.VirtualMachinePowerStateOff
+	case vimtypes.VirtualMachinePowerStateSuspended:
+		return vmopv1.VirtualMachinePowerStateSuspended
+	default:
+		logger.Info("Unknown snapshot power state, defaulting to Off",
+			"powerState", powerState)
+		return vmopv1.VirtualMachinePowerStateOff
+	}
+}
```

Comment:
Is `pkg/util/vmopv1/vm.go` a better place for this function since it's not specific to current vm snapshot file?